packages/core/nuxt-theme-module/theme/components/AppHeader.vue

```diff
@@ -1,37 +1,95 @@
-    </template>
-    <template #navigation>
-      <SfHeaderNavigationItem class="nav-item" data-cy="app-header-url_women" label="WOMEN" :link="localePath('/c/women')" />
-      <SfHeaderNavigationItem class="nav-item"  data-cy="app-header-url_men" label="MEN" :link="localePath('/c/men')" />
-      <SfHeaderNavigationItem class="nav-item" data-cy="app-header-url_kids" label="KIDS" :link="localePath('/c/kids')" />
-    </template>
-    <template #aside>
-      <LocaleSelector class="smartphone-only" />
-    </template>
-  </SfHeader>
+  <div>
+    <SfOverlay :visible="!!currentCategory" />
+    <SfHeader
+      data-cy="app-header"
+      @click:cart="toggleCartSidebar"
+      @click:wishlist="toggleWishlistSidebar"
+      @click:account="handleAccountClick"
+      @enter:search="changeSearchTerm"
+      @change:search="p => term = p"
+      :searchValue="term"
+      :cartItemsQty="cartTotalItems"
+      :accountIcon="accountIcon"
+      class="sf-header--has-mobile-search"
+    >
+      <!-- TODO: add mobile view buttons after SFUI team PR -->
+      <template #logo>
+        <nuxt-link data-cy="app-header-url_logo" :to="localePath('/')" class="sf-header__logo">
+          <SfImage src="/icons/logo.svg" alt="Vue Storefront Next" class="sf-header__logo-image"/>
+        </nuxt-link>
+      </template>
+      <template #navigation>
+        <SfHeaderNavigation
+          :is-visible-on-mobile="false"
+          @close="isVisible = false"
+        >
+          <SfHeaderNavigationItem
+            v-for="(category, index) in categories"
+            :key="index"
+            :label="category.name"
+            @mouseenter="currentCategory = category.name"
+            @mouseleave="currentCategory = ''"
+            @click="currentCategory = ''"
+            :link="localePath(`/c/${category.slug}`)"
+          >
+            <SfMegaMenu
+              :is-absolute="true"
+              :visible="currentCategory === category.name"
+              :title="category.name"
+              @close="currentCategory = ''"
+              class="sb-mega-menu"
+              v-if="category && category.children.length"
+            >
+              <SfMegaMenuColumn
+                v-for="(subcategory, subIndex) in category.children"
+                :key="subIndex"
+                :title="subcategory.name"
+              >
+                <SfList>
+                  <SfListItem
+                    v-for="(subcategoryChild, childIndex) in subcategory.children"
+                    :key="childIndex"
+                  >
+                    <SfMenuItem :label="subcategoryChild.name" :link="localePath(`/c/${subcategoryChild.slug}`)">
+                      <SfLink>
+                        {{ subcategoryChild.name }}
+                      </SfLink>
+                    </SfMenuItem>
+                  </SfListItem>
+                </SfList>
+              </SfMegaMenuColumn>
+              <SfMegaMenuColumn
+                v-if="currentCategory === 'New'"
```

Comment:
avoid plain strings and share this kind of stuff in variables and make it dynamic. Here you're assuming that every integration and shop has the same root categories